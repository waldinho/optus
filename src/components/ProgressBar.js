import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import * as style from '../styleVars/variables.js';

const Track = styled.div `
    ${style.width}
    height: 40px;
    background: ${style.yellow};
    border: 2px solid ${style.black};
`;

const Thumb = styled.div `
    width: ${props => props.percentage}%;
    max-width: 300px;
    height: 100%;
    background: ${props => props.limitColor};
    transition: width 0.3s ease-in-out;
`

const Percentage = styled.div `
    ${style.center}
    ${style.width}
    p {
        ${style.paragraph}
        font-weight: bold;
    }
`

const ProgressBar = ({bars, limit}) => {
    return (
        bars.map((item, i) => {
            const barPercentage = item < 0 ? 0 : item;
            const barPercentageLimit = barPercentage / (limit / 100);
            const limitColor = barPercentageLimit >= 100 ? `${style.red}` : `${style.green}`;
            return (
                <div key={i}>
                    <Percentage>
                        <p aria-label={`${barPercentageLimit.toFixed(0)}%`}>{barPercentageLimit.toFixed(0)}%</p>
                    </Percentage>
                    <Track>
                        <Thumb
                            percentage={barPercentageLimit}
                            limitColor={limitColor}
                        />
                    </Track>
                </div>
            )
        })
    )
}

ProgressBar.propTypes = {
    bars: PropTypes.array,
    limit: PropTypes.number.isRequired, 
}

ProgressBar.defaultProps = {
    bars: [],
    limit: 0, 
}

export default ProgressBar;