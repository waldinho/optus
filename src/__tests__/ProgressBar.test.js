import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import ProgressBar from '../components/ProgressBar';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ProgressBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const bars = [{
    0: 85,
    1: 65,
    2: 39
}];

const limit = 100;

describe('<ProgressBar /> with props', () => {
  const container = shallow(
      <ProgressBar 
        bars={bars}
        limit={limit}
      />
  );
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  })
})
