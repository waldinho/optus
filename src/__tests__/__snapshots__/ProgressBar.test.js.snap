// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> with props should match the snapshot 1`] = `"<div><div class=\\"sc-AxiKw gsNiXY\\"><p>NaN%</p></div><div class=\\"sc-AxjAm SrCId\\"><div class=\\"sc-AxirZ jWNpjQ\\"></div></div></div>"`;
